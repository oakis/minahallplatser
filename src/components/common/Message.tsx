import React from 'react';
import {View, StyleProp, TextStyle, ViewStyle} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {store} from '@src/App';
import {CLR_ERROR} from '@types';
import {Text} from '@common';
import {component} from '@style';

interface MessageProps {
  type: 'info' | 'success' | 'danger' | 'warning';
  message: string;
  backgroundColor: string;
}

const getIcon = (iconType: string): string => {
  switch (iconType) {
    case 'info':
      return 'info';
    case 'success':
      return 'check-circle';
    case 'danger':
      return 'error';
    case 'warning':
      return 'warning';
  }
  return '';
};

export const Message = ({
  type,
  message = '',
  backgroundColor,
}: MessageProps): JSX.Element => {
  if (message.length > 0) {
    return (
      <View
        style={
          [component.message.view, {backgroundColor}] as StyleProp<ViewStyle>
        }>
        {type ? (
          <Icon
            name={getIcon(type)}
            size={20}
            style={component.message.icon as StyleProp<TextStyle>}
          />
        ) : null}
        <Text style={component.message.text as StyleProp<TextStyle>}>
          {message}
        </Text>
        <Text
          style={{fontSize: 10, fontWeight: 'bold'}}
          onPress={() => store.dispatch({type: CLR_ERROR})}>
          StÃ¤ng
        </Text>
      </View>
    );
  }
  return <View />;
};
