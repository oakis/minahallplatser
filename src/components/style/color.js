export default {
    default: '#000',
    alternative: '#fff',
    primary: '#779ECB',
    info: '#FFA500',
    success: '#77DD77',
    danger: '#FF6961',
    warning: '#FFB347',
    
    lightgrey: '#efefef',
    darkgrey: '#dedede',
    smoothBlack: '#282828'
};
