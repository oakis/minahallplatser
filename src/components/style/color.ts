type ColorType =
  | 'default'
  | 'alternative'
  | 'primary'
  | 'primaryRGBA'
  | 'info'
  | 'success'
  | 'danger'
  | 'warning'
  | 'lightgrey'
  | 'darkgrey'
  | 'darkergrey'
  | 'smoothBlack'
  | 'overlay'
  | 'background';

type Color = {
  [key in ColorType]: string;
};

export const colors: Color = {
  default: '#000',
  alternative: '#fff',
  primary: '#779ECB',
  primaryRGBA: 'rgba(119, 158, 203, 0.4)',
  info: '#779ECB',
  success: '#77DD77',
  danger: '#FF6961',
  warning: '#FFB347',

  lightgrey: '#efefef',
  darkgrey: '#dedede',
  darkergrey: '#cecece',
  smoothBlack: '#282828',

  overlay: 'rgba(0, 0, 0, 0.4)',

  background: '#EAE9EF',
};
