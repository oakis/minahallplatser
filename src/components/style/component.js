import { Dimensions } from 'react-native';
import { colors, metrics } from './';

export const component = {
    button: {
        alignSelf: 'stretch',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        paddingTop: metrics.padding.md,
        paddingBottom: metrics.padding.md,
        marginBottom: metrics.margin.sm,
        borderRadius: metrics.borderRadius,
        elevation: 1,
        shadowRadius: 1,
        shadowColor: colors.smoothBlack,
        shadowOffset: { width: 0, height: 0 },
        shadowOpacity: 0.5
    },
    input: {
        container: {
            flexDirection: 'row',
            alignItems: 'center',
            marginBottom: 10,
            shadowRadius: 5,
            shadowColor: colors.smoothBlack,
            shadowOffset: { width: 0, height: 0 },
            shadowOpacity: 0.2
        }
    },
    text: {
        heading: {
            marginTop: metrics.margin.md,
            paddingLeft: metrics.padding.md,
            paddingRight: metrics.padding.sm,
            fontSize: 20,
            fontWeight: 'bold'
        },
        menu: {
            label: {
                fontSize: 12,
                fontWeight: 'bold',
                paddingLeft: metrics.padding.md,
                paddingTop: metrics.padding.md,
                paddingBottom: 0,
                marginBottom: 0
            },
            value: {
                paddingLeft: metrics.padding.md,
                marginBottom: metrics.margin.md
            }
        }
    },
    message: {
        view: {
            alignSelf: 'stretch',
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center',
            padding: metrics.padding.lg,
            margin: metrics.margin.sm,
            borderRadius: metrics.borderRadius,
            elevation: 5,
            shadowRadius: 5,
            shadowColor: colors.smoothBlack,
            shadowOffset: { width: 2.5, height: 2.5 },
            shadowOpacity: 0.5
        },
        text: {
            color: colors.alternative,
            fontSize: 12,
            alignSelf: 'center',
            flex: 1
        },
        icon: {
            marginRight: metrics.margin.lg,
            marginLeft: 0,
            marginTop: 2,
            alignSelf: 'center'
        }
    },
    listitem: {
        view: {
			flexDirection: 'row',
            justifyContent: 'space-between',
            alignItems: 'center',
            paddingHorizontal: metrics.padding.md,
			height: 50
		},
		text: {
			justifyContent: 'flex-start',
            alignSelf: 'center',
            fontSize: 18,
		},
		icon: {
            justifyContent: 'center',
            alignItems: 'center',
            width: 24,
		},
    },
    popup: {
        container: {
            position: 'absolute',
            zIndex: 1,
            height: '100%',
            width: '100%',
            justifyContent: 'center',
            alignItems: 'center',
            backgroundColor: colors.overlay
        },
        content: {
            position: 'absolute',
            width: '85%',
            minHeight: 250,
            maxHeight: (Dimensions.get('window').height * 0.9) / 1.3,
            zIndex: 2,
            backgroundColor: colors.alternative,
            borderRadius: 5,
            borderWidth: 2,
            elevation: 10,
            padding: metrics.padding.md
        },
        header: {
            fontSize: 18,
            fontWeight: 'bold',
            marginBottom: metrics.margin.sm
        },
        text: {
            marginBottom: metrics.margin.md
        },
        image: {
            marginBottom: metrics.margin.xl
        }
    }
};
