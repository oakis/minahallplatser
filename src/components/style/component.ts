import {Dimensions, StyleProp, TextStyle, ViewStyle} from 'react-native';
import {colors, metrics} from './';

export const component = {
  button: {
    alignSelf: 'stretch',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: metrics.padding.md,
    paddingBottom: metrics.padding.md,
    marginBottom: metrics.margin.sm,
    borderRadius: metrics.borderRadius,
    elevation: 1,
    shadowRadius: 1,
    shadowColor: colors.smoothBlack,
    shadowOffset: {width: 0, height: 0},
    shadowOpacity: 0.5,
  } as StyleProp<ViewStyle>,
  input: {
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      marginBottom: 10,
      shadowRadius: 5,
      shadowColor: colors.smoothBlack,
      shadowOffset: {width: 0, height: 0},
      shadowOpacity: 0.2,
    } as StyleProp<ViewStyle>,
  },
  text: {
    heading: {
      marginTop: metrics.margin.md,
      paddingLeft: metrics.padding.md,
      paddingRight: metrics.padding.sm,
      fontSize: 20,
      fontWeight: 'bold',
    } as StyleProp<TextStyle>,
    menu: {
      label: {
        fontSize: 12,
        fontWeight: 'bold',
        paddingLeft: metrics.padding.md,
        paddingTop: metrics.padding.md,
        paddingBottom: 0,
        marginBottom: 0,
      } as StyleProp<TextStyle>,
      value: {
        paddingLeft: metrics.padding.md,
        marginBottom: metrics.margin.md,
      } as StyleProp<TextStyle>,
    },
  },
  message: {
    view: {
      alignSelf: 'stretch',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      padding: metrics.padding.lg,
      margin: metrics.margin.sm,
      borderRadius: metrics.borderRadius,
      elevation: 5,
      shadowRadius: 5,
      shadowColor: colors.smoothBlack,
      shadowOffset: {width: 2.5, height: 2.5},
      shadowOpacity: 0.5,
    } as StyleProp<ViewStyle>,
    text: {
      color: colors.alternative,
      fontSize: 12,
      alignSelf: 'center',
      flex: 1,
    } as StyleProp<TextStyle>,
    icon: {
      marginRight: metrics.margin.lg,
      marginLeft: 0,
      marginTop: 2,
      alignSelf: 'center',
    } as StyleProp<TextStyle>,
  },
  listitem: {
    view: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      paddingHorizontal: metrics.padding.md,
      height: 50,
    } as StyleProp<ViewStyle>,
    text: {
      justifyContent: 'flex-start',
      alignSelf: 'center',
      fontSize: 18,
    } as StyleProp<TextStyle>,
    icon: {
      justifyContent: 'center',
      alignItems: 'center',
      width: 24,
    } as StyleProp<TextStyle>,
  },
  popup: {
    container: {
      position: 'absolute',
      zIndex: 1,
      height: '100%',
      width: '100%',
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: colors.overlay,
    } as StyleProp<ViewStyle>,
    content: {
      position: 'absolute',
      width: '85%',
      minHeight: 250,
      maxHeight: (Dimensions.get('window').height * 0.9) / 1.3,
      zIndex: 2,
      backgroundColor: colors.alternative,
      borderRadius: 5,
      borderWidth: 2,
      elevation: 10,
      padding: metrics.padding.md,
    } as StyleProp<ViewStyle>,
    header: {
      fontSize: 18,
      fontWeight: 'bold',
      marginBottom: metrics.margin.sm,
    } as StyleProp<TextStyle>,
    text: {
      marginBottom: metrics.margin.md,
    } as StyleProp<TextStyle>,
    image: {
      marginBottom: metrics.margin.xl,
    } as StyleProp<ViewStyle>,
  },
};
