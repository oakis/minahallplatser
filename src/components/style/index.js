export * from './color';
export * from './metrics';
export * from './component';
