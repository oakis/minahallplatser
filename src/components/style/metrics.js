export const metrics = {
    margin: {
        sm: 5,
        md: 10,
        lg: 20
    },
    padding: {
        sm: 5,
        md: 10,
        lg: 20
    },
    borderRadius: 3
};
