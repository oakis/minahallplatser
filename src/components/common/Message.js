import React from 'react';
import { View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { store } from '../../App';
import { CLR_ERROR } from '../../actions/types';
import { Text } from '../common';
import { colors, component } from '../style';

export const Message = ({ type, message = '' }) => {
    const getIcon = (iconType) => {
        switch (iconType) {
            case 'info':
                return 'ios-information-circle';
            case 'success':
                return 'ios-checkmark-circle';
            case 'danger':
                return 'ios-alert';
            case 'warning':
                return 'ios-warning';
            default:
                return;
        }
    };
    if (message.length > 0) {
        return (
            <View style={[component.message.view, { backgroundColor: colors[type] }]}>
                {(type) ? <Icon name={getIcon(type)} size={20} style={component.message.icon} /> : null}
                <Text style={component.message.text}>{message}</Text>
                <Text style={{ fontSize: 10, fontWeight: 'bold' }} onPress={() => store.dispatch({ type: CLR_ERROR })}>StÃ¤ng</Text>
            </View>
        );
    }
    return <View />;
};
