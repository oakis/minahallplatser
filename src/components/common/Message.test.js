import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import {store} from '../../App';
import {Message} from './Message';
import {colors} from '../style';

const types = ['info', 'success', 'danger', 'warning', null];

types.forEach(type => {
  it(`should match snapshot with type ${type}`, () => {
    const {toJSON} = render(
      <Message type={type} message="Message" backgroundColor={colors[type]} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});

it('should match snapshot with no message or type', () => {
  const {toJSON} = render(<Message />);
  expect(toJSON()).toMatchSnapshot();
});

it('should clear errors onPress', () => {
  store.dispatch = jest.fn();
  const {getByText} = render(<Message message="Message" />);
  fireEvent.press(getByText('St√§ng'));
  expect(store.dispatch).toBeCalledWith({type: 'CLR_ERROR'});
});
