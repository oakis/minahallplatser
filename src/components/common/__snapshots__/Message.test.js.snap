// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with no message or type 1`] = `<View />`;

exports[`should match snapshot with type danger 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 3,
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 5,
        "padding": 20,
        "shadowColor": "#282828",
        "shadowOffset": Object {
          "height": 2.5,
          "width": 2.5,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 5,
      },
      Object {
        "backgroundColor": "#FF6961",
      },
    ]
  }
>
  <MaterialIcon
    name="error"
    size={20}
    style={
      Object {
        "alignSelf": "center",
        "marginLeft": 0,
        "marginRight": 20,
        "marginTop": 2,
      }
    }
  />
  <DefaultFont
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "flex": 1,
        "fontSize": 12,
      }
    }
  >
    Message
  </DefaultFont>
  <DefaultFont
    onPress={[Function]}
    style={
      Object {
        "fontSize": 10,
        "fontWeight": "bold",
      }
    }
  >
    Stäng
  </DefaultFont>
</View>
`;

exports[`should match snapshot with type info 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 3,
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 5,
        "padding": 20,
        "shadowColor": "#282828",
        "shadowOffset": Object {
          "height": 2.5,
          "width": 2.5,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 5,
      },
      Object {
        "backgroundColor": "#779ECB",
      },
    ]
  }
>
  <MaterialIcon
    name="info"
    size={20}
    style={
      Object {
        "alignSelf": "center",
        "marginLeft": 0,
        "marginRight": 20,
        "marginTop": 2,
      }
    }
  />
  <DefaultFont
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "flex": 1,
        "fontSize": 12,
      }
    }
  >
    Message
  </DefaultFont>
  <DefaultFont
    onPress={[Function]}
    style={
      Object {
        "fontSize": 10,
        "fontWeight": "bold",
      }
    }
  >
    Stäng
  </DefaultFont>
</View>
`;

exports[`should match snapshot with type null 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 3,
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 5,
        "padding": 20,
        "shadowColor": "#282828",
        "shadowOffset": Object {
          "height": 2.5,
          "width": 2.5,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 5,
      },
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <DefaultFont
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "flex": 1,
        "fontSize": 12,
      }
    }
  >
    Message
  </DefaultFont>
  <DefaultFont
    onPress={[Function]}
    style={
      Object {
        "fontSize": 10,
        "fontWeight": "bold",
      }
    }
  >
    Stäng
  </DefaultFont>
</View>
`;

exports[`should match snapshot with type success 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 3,
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 5,
        "padding": 20,
        "shadowColor": "#282828",
        "shadowOffset": Object {
          "height": 2.5,
          "width": 2.5,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 5,
      },
      Object {
        "backgroundColor": "#77DD77",
      },
    ]
  }
>
  <MaterialIcon
    name="check-circle"
    size={20}
    style={
      Object {
        "alignSelf": "center",
        "marginLeft": 0,
        "marginRight": 20,
        "marginTop": 2,
      }
    }
  />
  <DefaultFont
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "flex": 1,
        "fontSize": 12,
      }
    }
  >
    Message
  </DefaultFont>
  <DefaultFont
    onPress={[Function]}
    style={
      Object {
        "fontSize": 10,
        "fontWeight": "bold",
      }
    }
  >
    Stäng
  </DefaultFont>
</View>
`;

exports[`should match snapshot with type warning 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 3,
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 5,
        "padding": 20,
        "shadowColor": "#282828",
        "shadowOffset": Object {
          "height": 2.5,
          "width": 2.5,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 5,
      },
      Object {
        "backgroundColor": "#FFB347",
      },
    ]
  }
>
  <MaterialIcon
    name="warning"
    size={20}
    style={
      Object {
        "alignSelf": "center",
        "marginLeft": 0,
        "marginRight": 20,
        "marginTop": 2,
      }
    }
  />
  <DefaultFont
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "flex": 1,
        "fontSize": 12,
      }
    }
  >
    Message
  </DefaultFont>
  <DefaultFont
    onPress={[Function]}
    style={
      Object {
        "fontSize": 10,
        "fontWeight": "bold",
      }
    }
  >
    Stäng
  </DefaultFont>
</View>
`;
